### PR TITLE
refactor: use Map in omitProps

### PR DESCRIPTION
**Pull request by mrm007**
## Description

Some files for `Modal` were named `Model` so I renamed them to `Modal`.

Replaced `toTrueHash` (which uses `.reduce` with `Object.assign` / `...` spread syntax) with `toMap` (which uses a native `Map`). `toTrueHash` is several orders of magnitude slower than the `Map` implementation and since it's on a critical path (every component uses `omitProps` I thought it would be a good idea to speed it up.

<details>
<summary>Details</summary>

```
toTrueHash x 4,855 ops/sec ±1.43% (83 runs sampled)
toMap x 156,257 ops/sec ±0.64% (88 runs sampled)
fromEntries x 39,484 ops/sec ±0.97% (88 runs sampled)
for_of x 210,762 ops/sec ±4.09% (82 runs sampled)
forEach x 140,859 ops/sec ±1.45% (84 runs sampled)
```

Run it: <https://gist.github.com/mrm007/d46afc18b51f4fa02d8bf434248b4fd0>

</details>

Let me know your thoughts!

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [x] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Angus](https://github.com/angusfretwell), [Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), or [Long](https://github.com/KieraDOG)).
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [x] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

File: src/components/omitProps/omitProps.js [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
 
-const styledSystemProps = toTrueHash(VALID_STYLED_SYSTEM_PROPS);
+const styledSystemProps = toMap(VALID_STYLED_SYSTEM_PROPS);
 
 const omitProps = (
   customPropsList = [],
   { omitStyledSystemProps = true } = {},
 ) => {
-  const customProps = toTrueHash(customPropsList);
+  const customProps = toMap(customPropsList); // ?
```
Comment (memcc) [inline src/components/omitProps/omitProps.js:12]: issue: let's remove the comment :)

**Review by memcc — APPROVED** [review]
Looks great mate!  Just one comment

File: src/components/omitProps/omitProps.js (hunk 1 above)
Comment (mrm007) [reply src/components/omitProps/omitProps.js]: Missed that one... https://quokkajs.com/docs/#comments

**Review by philipwindeyer — APPROVED** [review]
🚀

**Comment by ehyland** [comment]
👷 Deploy Preview for *roo-ui* processing.


🔨 Explore the source changes: ecc043a46f43b995e81063ea59023b07a99260c7

🔍 Inspect the deploy log: [https://app.netlify.com/sites/roo-ui/deploys/6111b2099752620008c49e3a](https://app.netlify.com/sites/roo-ui/deploys/6111b2099752620008c49e3a?utm_source=github&utm_campaign=bot_dl)

**Pull request merged**